File: Packages/HistoryKit/Sources/HistoryKit/HistoryEntry.swift
import Foundation

public enum HistoryKind: String, Codable, Equatable {
    case screenshot, recording
}

/// One remembered capture. Screenshots own a full-res PNG copy (`imageFile`);
/// recordings reference the user's saved file (`filePath`). Both own a JPEG
/// thumbnail. `imageFile`/`thumbFile` are names relative to the history
/// directory; `filePath` is absolute.
public struct HistoryEntry: Codable, Equatable, Identifiable {
    public let id: UUID
    public let kind: HistoryKind
    public let date: Date
    public let imageFile: String?
    public let filePath: String?
    public let thumbFile: String

    public init(id: UUID = UUID(), kind: HistoryKind, date: Date,
                imageFile: String? = nil, filePath: String? = nil, thumbFile: String) {
        self.id = id; self.kind = kind; self.date = date
        self.imageFile = imageFile; self.filePath = filePath; self.thumbFile = thumbFile
    }
}

File: windows/src/BetterScreenshot.History/HistoryEntry.cs
namespace BetterScreenshot.History;

public enum HistoryKind { Screenshot, Recording }

/// <summary>
/// One remembered capture. Screenshots are owned by history (a copied PNG + thumbnail); recordings are stored
/// by reference (the user's saved file path is never deleted by history). A thumbnail is always present.
/// </summary>
public sealed record HistoryEntry
{
    public Guid Id { get; init; } = Guid.NewGuid();
    public HistoryKind Kind { get; init; }
    public DateTime Date { get; init; }

    /// <summary>History-owned PNG filename (screenshots only).</summary>
    public string? ImageFile { get; init; }

    /// <summary>Absolute path to the user's saved recording (recordings only). Never deleted by history.</summary>
    public string? FilePath { get; init; }

    /// <summary>History-owned JPEG thumbnail filename. Always present.</summary>
    public string ThumbFile { get; init; } = string.Empty;
}
